code/modules/mob/living/carbon/slime/slime_extractor.dm

```diff
@@ -0,0 +1,121 @@
+	if(length(extract_slimes))
+		to_chat(user, FONT_SMALL(SPAN_WARNING("It is currently processing <b>[length(extract_slimes)] slime\s</b>.")))
+
+/obj/machinery/slime_extractor/update_icon()
+	cut_overlays()
+	if(panel_open)
+		var/mutable_appearance/panel_overlay = mutable_appearance(icon, "[icon_state]-panel")
+		add_overlay(panel_overlay)
+	if(length(extract_slimes))
+		var/mutable_appearance/interior_overlay = mutable_appearance(icon, "[icon_state]-interior", EFFECTS_ABOVE_LIGHTING_LAYER)
+		add_overlay(interior_overlay)
+		var/mutable_appearance/spinning_overlay = mutable_appearance(icon, "[icon_state]-running", EFFECTS_ABOVE_LIGHTING_LAYER)
+		add_overlay(spinning_overlay)
+		set_light(2.5, 1, COLOR_VIOLET)
+	else
+		set_light(FALSE)
+
+/obj/machinery/slime_extractor/RefreshParts()
+	..()
+	slime_limit = initial(slime_limit)
+	extraction_speed = initial(extraction_speed)
+
+	for(var/obj/item/stock_parts/P in component_parts)
+		if(ismatterbin(P))
+			slime_limit *= P.rating
+		else if(ismicrolaser(P))
+			extraction_speed = extraction_speed / P.rating
+
+/obj/machinery/slime_extractor/attackby(obj/item/O, mob/user)
+	if(length(extract_slimes))
+		to_chat(user, SPAN_WARNING("You can't modify \the [src] while it's busy. Please wait for the completion of previous operation."))
+		return
+
+	if(default_deconstruction_screwdriver(user, O))
+		return
+	if(default_deconstruction_crowbar(user, O))
+		return
+	if(default_part_replacement(user, O))
+		return
+
+/obj/machinery/slime_extractor/MouseDrop_T(atom/dropping, mob/user)
+	if(!Adjacent(user))
+		to_chat(user, SPAN_WARNING("You can't reach \the [src]!"))
+		return
+	if(istype(dropping, /mob/living/carbon/slime))
+		var/mob/living/carbon/slime/slimey = dropping
+		if(length(extract_slimes) >= slime_limit)
+			to_chat(user, SPAN_WARNING("\The [src] is fully loaded!"))
+			return
+		user.visible_message(SPAN_NOTICE("\The [user] starts loading \the [slimey] into \the [src]..."), SPAN_NOTICE("You start loading \the [slimey] into \the [src]..."))
+		if(do_after(user, 20))
+			if(!Adjacent(user))
+				to_chat(user, SPAN_WARNING("You can't reach \the [src]!"))
+				return
+			user.visible_message(SPAN_NOTICE("\The user loads \the [slimey] into \the [src]."), SPAN_NOTICE("You load \the [slimey] into \the [src]."))
+			slimey.forceMove(src)
+			extract_slimes[slimey] = slimey
+			addtimer(CALLBACK(src, .proc/extraction_process, slimey), extraction_speed)
+			update_icon()
+
+/obj/machinery/slime_extractor/proc/extraction_process(var/slime)
+	if(!length(extract_slimes))
+		update_icon()
+		return
+	var/mob/living/carbon/slime/extracted_slime = extract_slimes[slime]
+	if(!extracted_slime)
+		update_icon()
+		return
+	for(var/i = 1 to extracted_slime.cores + 1)
+		var/obj/extract = new extracted_slime.coretype(get_turf(src))
+		var/obj/item/storage/bag/slimes/slime_bag = locate() in range(1, get_turf(src))
+		if(slime_bag && Adjacent(slime_bag))
```

Comment:
```suggestion
		if(slime_bag)
```

We already know it is Adjacent, we just did range(1, ...)

Reply:
Adjacent is to check for stuff like windows in the way, not just range